src/modules/mavlink/mavlink_main.h

```diff
@@ -426,7 +427,15 @@ class Mavlink
 
 	ringbuffer::RingBuffer	*get_logbuffer() { return &_logbuffer; }
 
-	unsigned		get_system_type() { return _system_type; }
+	unsigned		get_system_type()
```

Comment:
Maybe leave it alone here print an error in the param update? You could also set the param to 0 there.